src/value.c

```diff
@@ -371,58 +371,59 @@ SIValue SIValue_Divide(const SIValue a, const SIValue b) {
 	return SI_DoubleVal(SI_GET_NUMERIC(a) / (double)SI_GET_NUMERIC(b));
 }
 
-int SIArray_Compare(SIValue arrayA, SIValue arrayB) {
+int SIArray_Compare(SIValue arrayA, SIValue arrayB, int *disjointOrNull) {
 	uint arrayALen = SIArray_Length(arrayA);
 	uint arrayBLen = SIArray_Length(arrayB);
-	// check empty list
+	// Check empty list.
 	if(arrayALen == 0 && arrayBLen == 0) return 0;
 	int lenDiff = arrayALen - arrayBLen;
-	// check for the common range of indices
-	uint minLengh = arrayALen <= arrayBLen ? arrayALen : arrayBLen;
+	// Check for the common range of indices.
+	uint minLength = arrayALen <= arrayBLen ? arrayALen : arrayBLen;
 
-	uint allDisjoint = 0;   // counter for the amount of disjoint comparisons
-	uint nullCompare = 0;   // counter for the amount of null comparison
-	// notEqual holds the first false (result != 0) comparison result between two values from the same type, which are not equal
+	uint notEqualCounter = 0;   // Counter for the amount of false (compare(a,b) !=0) comparisons.
```

Comment:
Reorder `notEqualCounter`, `nullCounter` and `notEqual`
by line length such that `notEqual` appears first.

Reply:
done